File: lib_foundation/src/main/java/com/android/base/app/activity/ActivityDelegateOwner.java
package com.android.base.app.activity;


import com.github.dmstocking.optional.java.util.function.Predicate;

import androidx.annotation.UiThread;

@UiThread
@SuppressWarnings("unused")
public interface ActivityDelegateOwner {

    void addDelegate(ActivityDelegate fragmentDelegate);

    boolean removeDelegate(ActivityDelegate fragmentDelegate);

    ActivityDelegate findDelegate(Predicate<ActivityDelegate> predicate);

    ActivityState getStatus();

}

File: lib_foundation/src/main/java/com/android/base/foundation/activity/ActivityDelegateOwner.kt
package com.android.base.foundation.activity

import androidx.annotation.UiThread

@UiThread
interface ActivityDelegateOwner {

    fun addDelegate(activityDelegate: ActivityDelegate<*>)

    fun removeDelegate(activityDelegate: ActivityDelegate<*>): Boolean

    fun findDelegate(predicate: (ActivityDelegate<*>) -> Boolean): ActivityDelegate<*>?

    fun getStatus(): ActivityState

}